CI failure in integration tests:

```
Traceback (most recent call last):
  File "/tmp/ansible_lJH5KV/ansible_module_synchronize.py", line 526, in <module>
    main()
  File "/tmp/ansible_lJH5KV/ansible_module_synchronize.py", line 416, in main
    private_key = os.path.expanduser(private_key)
  File "/usr/lib/python2.7/posixpath.py", line 254, in expanduser
    if not path.startswith('~'):
AttributeError: 'NoneType' object has no attribute 'startswith'
```

--- a/lib/ansible/modules/files/synchronize.py
+++ b/lib/ansible/modules/files/synchronize.py
@@ -467,7 +467,7 @@ def main():
     if is_rsh_needed(source, dest):
         ssh_cmd = [module.get_bin_path('ssh', required=True), '-S', 'none']
         if private_key is not None:
-            ssh_cmd.extend(['-i', private_key])
+            ssh_cmd.extend(['-i', os.path.expanduser(private_key) ])
         # If the user specified a port value
         # Note:  The action plugin takes care of setting this to a port from
         # inventory if the user didn't specify an explicit dest_port

--- a/lib/ansible/plugins/action/synchronize.py
+++ b/lib/ansible/plugins/action/synchronize.py
@@ -317,6 +317,8 @@ class ActionModule(ActionBase):
                 if use_delegate:
                     user = task_vars.get('ansible_delegated_vars', dict()).get('ansible_ssh_user', None)
                     if not user:
+                        user = task_vars.get('ansible_ssh_user') or self._play_context.remote_user
+                    if not user:
                         user = C.DEFAULT_REMOTE_USER
 
                 else:
@@ -326,7 +328,6 @@ class ActionModule(ActionBase):
             private_key = self._play_context.private_key_file
 
             if private_key is not None:
-                private_key = os.path.expanduser(private_key)
                 _tmp_args['private_key'] = private_key
 
             # use the mode to define src and dest's url

--- a/test/units/plugins/action/fixtures/synchronize/delegate_remote/meta.yaml
+++ b/test/units/plugins/action/fixtures/synchronize/delegate_remote/meta.yaml
@@ -23,4 +23,4 @@ asserts:
     - "self.execute_called"
     - "self.final_module_args['_local_rsync_path'] == 'rsync'"
     - "self.final_module_args['src'] == '/tmp/deleteme'"
-    - "self.final_module_args['dest'] == 'el6host:/tmp/deleteme'"
+    - "self.final_module_args['dest'] == 'root@el6host:/tmp/deleteme'"

--- a/test/units/plugins/action/fixtures/synchronize/delegate_remote_su/meta.yaml
+++ b/test/units/plugins/action/fixtures/synchronize/delegate_remote_su/meta.yaml
@@ -30,4 +30,4 @@ asserts:
     - "self.execute_called"
     - "self.final_module_args['_local_rsync_path'] == 'rsync'"
     - "self.final_module_args['src'] == '/tmp/deleteme'"
-    - "self.final_module_args['dest'] == 'el6host:/tmp/deleteme'"
+    - "self.final_module_args['dest'] == 'root@el6host:/tmp/deleteme'"

